apps/token-manager/app/src/app-logic.js

```diff
@@ -0,0 +1,188 @@
+import React, { useCallback, useMemo, useState, useEffect } from 'react'
+import BN from 'bn.js'
+import {
+  useAppState,
+  useCurrentApp,
+  useInstalledApps,
+  usePath,
+} from '@aragon/api-react'
+import { addressesEqual } from './web3-utils'
+
+const HOLDER_ADDRESS_PATH = /^\/vesting\/0x[a-fA-F0-9]{40}\/?$/
+const NO_HOLDER_ADDRESS = '-1'
+
+function holderFromPath(path) {
+  if (!path) {
+    return NO_HOLDER_ADDRESS
+  }
+  const matches = path.match(HOLDER_ADDRESS_PATH)
+  if (matches) {
+  }
+  return matches ? matches[0].split('/')[2] : NO_HOLDER_ADDRESS
+}
+
+// Get the vestings from the holder currently selected, or null otherwise.
+export function useSelectedHolderVestings() {
+  const { vestings, holders } = useAppState()
+  const [path, requestPath] = usePath()
+  let holderVestingInfo = {}
+
+  // The memoized holder currently selected.
+  const selectedHolder = useMemo(() => {
+    const holderAddress = holderFromPath(path)
+
+    if (holderAddress === NO_HOLDER_ADDRESS) {
+      return null
+    }
+
+    if (vestings) {
+      holderVestingInfo = vestings.find(
+        vesting => vesting.receiver === holderAddress
+      )
+    }
+    if (holders) {
+      holderVestingInfo.holderBalance = holders.find(
+        holder => holder.address === holderAddress
+      )
+    }
+    return holderVestingInfo
+  }, [path, vestings])
+
+  const selectHolder = useCallback(
+    holderAddress => {
+      requestPath(
+        String(holderAddress) === NO_HOLDER_ADDRESS
+          ? ''
+          : `/vesting/${holderAddress}/`
+      )
+    },
+    [requestPath]
+  )
+
+  return [selectedHolder, selectHolder]
+}
+
+// Decorate the vestings array with more information relevant
+function useDecoratedVestings() {
+  const { vestings, holders } = useAppState()
+  const currentApp = useCurrentApp()
+  const installedApps = useInstalledApps()
+
+  return useMemo(() => {
+    // Make sure we have loaded information about the current app and other installed apps before showing votes
+    if (!(vestings && currentApp && installedApps)) {
+      return [[], []]
+    }
+
+    return vestings
+  }, [vestings])
+}
+
+export function toISODate(seconds) {
```

Comment:
To be consistent with the other apps, we should be transforming the times in the script.js to their unix time equivalents in ms.

You can find examples in Voting's script, where we use a `marshall()` function to do the transformations.